doc/R-vignette/igraph_extended_tutorial_R_vignette_english.Rmd

```diff
@@ -0,0 +1,621 @@
+
+We now have an directed graph with 6 vertices and 7 edges. Vertex and edge IDs are always *continuous*, so if you delete a vertex all subsequent vertices will be renumbered. When a vertex is renumbered, edges are **not** renumbered, but their source and target vertices will. Use `Graph.delete_vertices` and `Graph.delete_edges` to perform these operations. For instance, to delete the edge connecting vertices `3-4`, get its ID and then delete it:
+
+```{r echo = TRUE}
+get.edge.ids(gi, c(3,4))
+```
+
+```{r}
+gi <- delete.edges(gi, 4)
+```
+
+## Generating graphs
+
+igraph includes both deterministic and stochastic graph generators (see `generation`). *Deterministic* generators produce the same graph every time you call the fuction, e.g.:
+
+```{r echo = TRUE}
+gi <- make_empty_graph()
+gi <- make_tree(127, 2, mode = "undirected")
+summary(gi)
+```
+
+Uses `make_tree` to generate a regular tree graph with 127 vertices, each vertex having two children (and one parent, of course). No matter how many times you call `make_tree`, the generated graph will always be the same if you use the same parameters:
+
+```{r}
+gi2 <- make_tree(127, 2, mode = "undirected")
+```
+
+```{r echo = TRUE}
+get.edgelist(gi) == get.edgelist(gi2)
+```
+
+The above code snippet also shows you that the `get_edgelist()` method, which returns a list of source and target vertices for all edges, sorted by edge ID. If you print the first 10 elements, you get:
+
+```{r echo = TRUE}
+head(get.edgelist(gi2),10)
+```
+
+*Stochastic* generators produce a different graph each time, e.g. `sample_grg`:
+
+```{r echo = TRUE}
+gi <- make_empty_graph()
+gi <- sample_grg(100, 0.2)
+summary(gi)
+```
+
+This generates a geometric random graph: *n* points are chosen randomly and uniformly inside the unit square and pairs of points closer to each other than a predefined distance *d* are connected by an edge. If you generate GRGs with the same parameters, they will be different:
+
+```{r echo = TRUE}
+gi2 <- make_empty_graph()
+gi2 <- sample_grg(100, 0.2)
+
+gi_a <- head(get.edgelist(gi), 10) 
+gi2_a <-head(get.edgelist(gi2), 10)
+
+gi_a == gi2_a
+```
+
+A slightly looser way to check if the graphs are equivalent is via `isomorphic()`:
+
+```{r echo = TRUE}
+isomorphic(gi,gi2)
+```
+
+Checking for isomorphism can take a while for large graphs (in this case, the answer can quickly be given by checking the degree distributions of the two graphs).
+
+## Setting and retrieving attributes
+
+As mentioned above, in igraph each vertex and each edge have a numeric id from `1` upwards. Deleting vertices or edges can therefore cause reassignments of vertex and/or edge IDs. In addition to IDs, vertex and edges can have *attributes* such as a name, coordinates for plotting, metadata, and weights. The graph itself can have such attributes too (e.g. a name, which will show in `print` or `summary`). In a sense, every `Graph`, vertex and edge can be used as a R variable to store and retrieve these attributes.
```

Comment:
Please use "id" vs "ID" consistently throughout the tutorial.

Given the ID reassignments after vertex deletion, it would be nice to introduce vertex names much earlier, even before discussing attributes, and demonstrating how vertices can be referred to either by string name or integer ID.